airflow/providers/amazon/aws/operators/bedrock.py

```diff
@@ -0,0 +1,89 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import json
+from typing import TYPE_CHECKING, Any, Sequence
+
+from airflow.providers.amazon.aws.hooks.bedrock import BedrockRuntimeHook
+from airflow.providers.amazon.aws.operators.base_aws import AwsBaseOperator
+from airflow.utils.helpers import prune_dict
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+DEFAULT_CONN_ID = "aws_default"
+
+
+class BedrockInvokeModelOperator(AwsBaseOperator[BedrockRuntimeHook]):
+    """
+    Invoke the specified Bedrock model to run inference using the input provided.
+
+    Use InvokeModel to run inference for text models, image models, and embedding models.
+    To see the format and content of the input_data field for different models, refer to
+    `Inference parameters docs <https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters.html>`_.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:BedrockInvokeModelOperator`
+
+    :param model_id: The ID of the Bedrock model. (templated)
+    :param input_data: Input data in the format specified in the content-type request header. (templated)
+    :param content_type: The MIME type of the input data in the request. (templated) Default: application/json
+    :param accept: The desired MIME type of the inference body in the response.
+        (templated) Default: application/json
+
+    :param aws_conn_id: The Airflow connection used for AWS credentials.
+        If this is ``None`` or empty then the default boto3 behaviour is used. If
+        running Airflow in a distributed manner and aws_conn_id is None or
+        empty, then default boto3 configuration would be used (and must be
+        maintained on each worker node).
+    :param region_name: AWS region_name. If not specified then the default boto3 behaviour is used.
```

Comment:
should we add defer support for this operator?

Reply:
IMHO, there is no any benefit of deferral operator right now, there is single call to the single api endpoint and return response. It would be some benefit, if for example there is two calls, one send response, and other wait until it ready

Reply:
I assumed we want also to allow waiting for completion :)

Reply:
Sadly, we can't on this one.  The API call doesn't return until the task completes.  You can see what I mean if you run the following in ipython or whatever python console you like:

```
import boto3

boto3.client("bedrock-runtime").invoke_model(body='{"prompt": "What are the first 10 digits of pi?"}', modelId='meta.llama2-13b-chat-v1')
```

It sends the request then waits for the task to complete before returning.

Reply:
Sorry, I don't know why the rest of that convo didn't show up.  @Taragolis is right, it behaves similar to a lambda call, unfortunately.   When I add the operators for customizeModel and provisionThroughput, those I should be able to make deferrable.

Reply:
Cool!